src/cli/java/org/commcare/util/screen/EntityListSubscreen.java

```diff
@@ -236,10 +239,24 @@ public boolean handleInputAndUpdateHost(String input, EntityScreen host, boolean
 
         if (handleCaseIndex) {
             try {
-                int index = Integer.parseInt(input);
-                TreeReference[] selectedRefs = new TreeReference[1];
-                selectedRefs[0] = mChoices[index];
-                host.updateSelection(input, selectedRefs);
+                if (input.contains(",")) {
+                    String[] inputArray = input.split(",");
+                    String[] inputValues = new String[inputArray.length];
+                    for (int i = 0; i < inputArray.length; i++) {
+                        int currentIndex = Integer.parseInt(inputArray[i]);
+                        inputValues[i] = host.getReturnValueFromSelection(mChoices[currentIndex]);
+                    }
+                    host.updateSelection(USE_SELECTED_VALUES, inputValues);
+                } else {
+                    int index = Integer.parseInt(input);
+                    TreeReference[] selectedRefs = new TreeReference[1];
+                    selectedRefs[0] = mChoices[index];
+                    if (host instanceof MultiSelectEntityScreen) {
+                        host.updateSelection(USE_SELECTED_VALUES, selectedRefs);
+                    } else {
+                        host.updateSelection(input, selectedRefs);
+                    }
+                }
```

Comment:
In correlation to my comment above on  semantics of `input` and `use_selected_values`, this should get modified as below to handle input correctly for multi-select. 
```suggestion              
                TreeReference[] selectedRefs = null;
                if (host instanceof MultiSelectEntityScreen) {
                    if (input.contentEquals(USE_SELECTED_VALUES)) {
                        if (selectedValues == null) {
                            throw new IllegalArgumentException("selected values can't be null");
                        }
                        selectedRefs = new TreeReference[selectedValues.length];
                        for (int i = 0; i < selectedValues.length; i++) {
                            int index = Integer.parseInt(selectedValues[i]);
                            selectedRefs[i] = mChoices[index];
                        }
                    }
                } else {
                    int index = Integer.parseInt(input);
                    selectedRefs = new TreeReference[1];
                    selectedRefs[0] = mChoices[index];
                }
                host.updateSelection(input, selectedRefs);   
```